sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -26,40 +29,428 @@ class ConfigurationSetting(Model):
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+
+    def __init__(self, key, enabled, feature_filters=None, **kwargs):
+        # type: (str, bool, Optional[List[FeatureFilterBase]]) -> None
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        if not key.startswith(self.key_prefix):
+            self.key = self.key_prefix + key
+        self.enabled = enabled
+        self.label = kwargs.get("label", None)
+        self.content_type = kwargs.get("content_type", self.feature_flag_content_type)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.tags = kwargs.get("tags", {})
+        self.read_only = kwargs.get("read_only", None)
+        self.etag = kwargs.get("etag", None)
+        self.description = kwargs.get("description", None)
+        self.display_name = kwargs.get("display_name", None)
+        self.feature_filters = feature_filters or []
+
+    @classmethod
+    def _from_generated(cls, key_value):
+        # type: (KeyValue) -> FeatureFlagConfigurationSetting
+        if key_value is None:
+            return None
+        if key_value.value:
+            try:
+                key_value.value = json.loads(key_value.value)
+            except json.decoder.JSONDecodeError:
+                pass
+
+        filters = None
+        try:
+            filters = key_value.value["conditions"]["client_filters"]
+            if filters == [None]:
+                key_value.value["conditions"]["client_filters"] = []
+            if len(filters) > 0 and filters != [None]:
+                filters = [
+                    FeatureFilterBase._from_generated(f) for f in filters  # pylint: disable=protected-access
+                ]  # pylint: disable=protected-access
+                key_value.value["conditions"]["client_filters"] = filters
+        except KeyError:
+            pass
+
+        return cls(
+            key=key_value.key,
+            enabled=key_value.value["enabled"],
+            label=key_value.label,
+            content_type=key_value.content_type,
+            last_modified=key_value.last_modified,
+            tags=key_value.tags,
+            read_only=key_value.locked,
+            etag=key_value.etag,
+            feature_filters=filters,
+        )
+
+    def _to_generated(self):
+        # type: (...) -> KeyValue
+        value = {
```

Comment:
Won't this overwrite the customer's value if it's not well-formed JSON?

Reply:
No, this does not overwrite the customer's value. `key` and `description` should be strings, and `enabled` should be a boolean so there is no fear of invalid json

Reply:
The service doesn't validate the value of the `value` property to be JSON at all even if the content type and the prefix are both set to represent a feature flag. So imagine you get `{ "value":"3" }` from the service. When re-serializing this code would replace it with the full feature-flag definitions `{ "value": "{ \"isEnabled\":false..... }" }` losing the original.